src/explorer/SiteTreeItem.ts

```diff
@@ -212,21 +213,22 @@ export abstract class SiteTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     }
 
-    private getIgnoredFoldersForDeployment(runtime: string): string[] {
+    private async getIgnoredFoldersForDeployment(fsPath: string, runtime: string): Promise<string[]> {
         let ignoredFolders: string[];
         switch (runtime) {
             case LinuxRuntimes.node:
                 ignoredFolders = ['node_modules{,/**}'];
                 break;
             case LinuxRuntimes.python:
+                const venvsfsPaths: string[] = (await venvUtils.getExistingVenvs(fsPath)).map(venvPath => `${venvPath}{,/**}`);
```

Comment:
If this fails (maybe user doesn't have access to a directory?) I feel like we should catch that rather than block deploy